File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.cpp
class Solution {
public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        int leftMax = 0;
        for(int i=0; i<left.size(); i++){
            leftMax = max(leftMax, left[i]);
        }

        int rightMin = n;
        for(int i=0; i<right.size(); i++){
            rightMin = min(rightMin, right[i]);
        }

        return max(leftMax, n-rightMin);
    }
};

File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        int leftMax = 0;

        for (int i = 0; i < left.length; i++) {
            leftMax = Math.max(leftMax, left[i]);
        }

        int rightMin = n;

        for (int i = 0; i < right.length; i++) {
            rightMin = Math.min(rightMin, right[i]);
        }

        return Math.max(leftMax, n - rightMin);
    }
}